internal/pkg/manifest/validate.go

```diff
@@ -0,0 +1,410 @@
+		}
+	}
+	return nil
+}
+
+// Validate returns if Range is configured correctly.
+func (r *Range) Validate() error {
+	if !r.RangeConfig.IsEmpty() {
+		return r.RangeConfig.Validate()
+	}
+	return r.Value.Validate()
+}
+
+// Validate returns if IntRangeBand is configured correctly.
+func (r *IntRangeBand) Validate() error {
+	str := string(*r)
+	minMax := intRangeBandRegexp.FindStringSubmatch(str)
+	// Valid minMax example: ["1-2", "1", "2"]
+	if len(minMax) != 3 {
+		return fmt.Errorf("invalid range value %s. Should be in format of ${min}-${max}", str)
+	}
+	// Guaranteed by intRangeBandRegexp.
+	min, err := strconv.Atoi(minMax[1])
+	if err != nil {
+		return err
+	}
+	max, err := strconv.Atoi(minMax[2])
+	if err != nil {
+		return err
+	}
+	if min <= max {
+		return nil
+	}
+	return &errMinGreaterThanMax{
+		min: min,
+		max: max,
+	}
+}
+
+// Validate returns if RangeConfig is configured correctly.
+func (r *RangeConfig) Validate() error {
+	min, max := aws.IntValue(r.Min), aws.IntValue(r.Max)
+	if min <= max {
+		return nil
+	}
+	return &errMinGreaterThanMax{
+		min: min,
+		max: max,
+	}
+}
+
+// Validate returns if ExecuteCommand is configured correctly.
+func (e *ExecuteCommand) Validate() error {
+	if !e.Config.IsEmpty() {
+		return e.Config.Validate()
+	}
+	return nil
+}
+
+// Validate returns if ExecuteCommandConfig is configured correctly.
+func (*ExecuteCommandConfig) Validate() error {
+	return nil
+}
+
+// Validate returns if Storage is configured correctly.
+func (s *Storage) Validate() error {
+	for k, v := range s.Volumes {
+		if err := v.Validate(); err != nil {
+			return fmt.Errorf(`validate "volumes[%s]": %w`, k, err)
+		}
+	}
+	return nil
+}
+
+// Validate returns if Volume is configured correctly.
+func (v *Volume) Validate() error {
+	if err := v.EFS.Validate(); err != nil {
+		return fmt.Errorf(`validate "efs": %w`, err)
+	}
+	return v.MountPointOpts.Validate()
+}
+
+// Validate returns if MountPointOpts is configured correctly.
+func (*MountPointOpts) Validate() error {
+	return nil
+}
+
+// Validate returns if EFSConfigOrBool is configured correctly.
+func (e *EFSConfigOrBool) Validate() error {
+	if !e.Advanced.IsEmpty() {
+		return e.Advanced.Validate()
+	}
+	return nil
+}
+
+// Validate returns if EFSVolumeConfiguration is configured correctly.
+func (e *EFSVolumeConfiguration) Validate() error {
+	if !e.EmptyBYOConfig() && !e.EmptyUIDConfig() {
+		return &errFieldMutualExclusive{
+			firstField:  "uid/gid",
+			secondField: "id/root_dir/auth",
+		}
+	}
+	if e.UID != nil && e.GID == nil {
+		return &errFieldMustBeSpecified{
+			missingField:      "gid",
+			conditionalFields: []string{"uid"},
+		}
+	}
+	if e.UID == nil && e.GID != nil {
+		return &errFieldMustBeSpecified{
+			missingField:      "uid",
+			conditionalFields: []string{"gid"},
+		}
+	}
+	if err := e.AuthConfig.Validate(); err != nil {
+		return fmt.Errorf(`validate "auth": %w`, err)
+	}
+	if aws.StringValue(e.AuthConfig.AccessPointID) != "" {
```

Comment:
I wonder if `aws.StringValue(e.AuthConfig.AccessPointID) != ""` could ever be true. What would the `manifest.yml` be like that `AccessPointID` is marshaled as `""`?

Same for `RootDirectory`.